### PR TITLE
Add agent test

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: cmd/agent/agent_test.go [hunk 1]
```diff
@@ -17,71 +24,259 @@ func TestT(t *testing.T) {
...
+
+	s.callDashboardAPI(c, nil, "/api/slot", "POST", v)
+}
+
+func (s *testAgentSuite) testProxy(c *C) {
+	agent := s.agentProxy
+	proxyAddr := "127.0.0.1:19000"
+	proxyHTTPAddr := "127.0.0.1:29000"
+
+	agent.httpCall(c, nil, "start_proxy", fmt.Sprintf("addr=%s&http_addr=%s", proxyAddr, proxyHTTPAddr), "POST")
```
Comment (qiuyesuifeng) [inline cmd/agent/agent_test.go]: is it ok to parse a "127.0.0.1:19000" into http arg? maybe use url encode?

Comment (siddontang) [inline cmd/agent/agent_test.go]: Yes, I will change it later.

**Comment by qiuyesuifeng** [comment]
LGTM.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**